include/xtensor/xoperation.hpp

```diff
@@ -959,6 +959,74 @@ namespace xt
     {
         return detail::make_xfunction<typename detail::cast<R>::functor>(std::forward<E>(e));
     }
+
+    /**
+    * @ingroup logical_operators
+    * @brief isin
+    *
+    * Returns a boolean array of the same shape as element that is ``true`` where an element of
+    * ``element`` is in ``test_elements`` and ``False`` otherwise.
+    * @param e an \ref xexpression
+    * @return a boolean array
+    */
+    template <class E, class T>
+    inline auto isin(E&& e, std::initializer_list<T> values) noexcept
+    {
+        auto lambda = [values](const auto& t) -> decltype(e == e) {
+            return std::find(values.begin(), values.end(), t) != values.end(); };
+        return make_lambda_xfunction(std::move(lambda), std::forward<E>(e));
+    }
+
+    /**
+    * @ingroup logical_operators
+    * @brief isin
+    *
+    * Returns a boolean array of the same shape as element that is ``true`` where an element of
+    * ``element`` is in ``test_elements`` and ``False`` otherwise.
+    * @param e an \ref xexpression
+    * @return a boolean array
+    */
+    template <class E, class F>
+    inline auto isin(E&& e, F&& values) noexcept
```

Comment:
I wonder if we should not accept a pair of iterators here instead of a universal reference. If we stick with universal reference, it could be nice to disable this function when the type `F` does not provide `begin` / `end` methods.

Reply:
Maybe I did not get you right, but I think that `values` should, in my opinion, at least directly  accept an `xtensor` and maybe also an `std::vector` (i.e. without passing iterators). I have no problems also allowing iterators.

Reply:
I think it is a good idea to disable this function when it does not provide `begin` and `end`

Reply:
Accepting pair of iterators makes it a bit more C++-ish, and more flexible (the user can pass any kind of iterators, even pseudo iterators). However I agree that passing directly a container is convenient. So I would say let's add them both (the overload taking a pair of iterators can be added later though).

Reply:
Let's to it in this PR too!

Reply:
It should be there now!